Framework/Core/include/Framework/ASoA.h

```diff
@@ -582,7 +582,7 @@ struct IndexPolicyBase {
 };
 
 struct RowViewSentinel {
-  uint64_t const index;
+  int64_t const index;
```

Comment:
Please, don't tell me this was the issue...

Reply:
It was not, but it was inconsistent with the index values in other iterators.

Reply:
However, since cannot declare `operator <=> = default`, we also have to declare `operator==` explicitly. And while we an define ordering on the set of iterators, we do not need it for anything.